src/shogun/machine/gp/GaussianARDFITCKernel.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) The Shogun Machine Learning Toolbox
+ * Written (w) 2015 Wu Lin
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation are those
+ * of the authors and should not be interpreted as representing official policies,
+ * either expressed or implied, of the Shogun Development Team.
+ *
+ */
+
+#include <shogun/machine/gp/GaussianARDFITCKernel.h>
+
+using namespace shogun;
+
+CGaussianARDFITCKernel::CGaussianARDFITCKernel() : CGaussianARDKernel()
+{
+	init();
+}
+void CGaussianARDFITCKernel::init()
+{
+}
+
+CGaussianARDFITCKernel::~CGaussianARDFITCKernel()
+{
+}
+
+#ifdef HAVE_CXX11
+#include <shogun/mathematics/linalg/linalg.h>
+CGaussianARDFITCKernel::CGaussianARDFITCKernel(int32_t size, float64_t width)
+		: CGaussianARDKernel(size,width)
+{
+	init();
+}
+
+CGaussianARDFITCKernel::CGaussianARDFITCKernel(CDotFeatures* l,
+		CDotFeatures* r, int32_t size, float64_t width)
+		: CGaussianARDKernel(l, r, size, width)
+{
+	init();
+}
+
+CGaussianARDFITCKernel* CGaussianARDFITCKernel::obtain_from_generic(CKernel* kernel)
+{
+	if (kernel->get_kernel_type()!=K_GAUSSIANARDFITC)
+	{
+		SG_SERROR("Provided kernel is not of type CGaussianARDFITCKernel!\n");
+	}
+
+	/* since an additional reference is returned */
+	SG_REF(kernel);
+	return (CGaussianARDFITCKernel*)kernel;
+}
+
+#include <typeinfo>
+CSGObject *CGaussianARDFITCKernel::shallow_copy() const
+{
+	// TODO: remove this after all the classes get shallow_copy properly implemented
```

Comment:
done 
see
https://github.com/shogun-toolbox/shogun/issues/2735
For now, I will remove this function.